packages/contracts/test/diamond/DiamondTestSetup.sol

```diff
@@ -0,0 +1,186 @@
+        });
+
+        cut[1] = FacetCut({
+            facetAddress: address(ownerF),
+            action: FacetCutAction.Add,
+            functionSelectors: generateSelectors("OwnershipFacet")
+        });
+
+
+        // initialise interfaces
+        ILoupe = IDiamondLoupe(address(diamond));
+        ICut = IDiamondCut(address(diamond));
+
+        //upgrade diamond
+        vm.startPrank(owner);
+        ICut.diamondCut(cut, address(0x0), "");
+		vm.stopPrank();
+
+        // get all addresses
+        facetAddressList = ILoupe.facetAddresses();
+    }
+}
+
+// tests proper upgrade of diamond when adding a facet
+abstract contract AddManagerFacetSetup is DiamondSetup {
+
+    ManagerFacet managerFacet;
+
+    function setUp() public virtual override {
+        super.setUp();
+        //deploy ManagerFacet
+        managerFacet = new ManagerFacet(owner);
+
+        // bytes4[] memory fromGenSelectors  = generateSelectors("ManagerFacet"); // all length is 51
+        bytes4[] memory selectorsInManagerFacet = new bytes4[](6);
+        selectorsInManagerFacet[0] = getSelector(
+            "setTwapOracleAddress(address)"
+        );
+        selectorsInManagerFacet[1] = getSelector(
+            "setuARTokenAddress(address)"
+        );
+        selectorsInManagerFacet[2] = getSelector(
+            "setDebtCouponAddress(address)"
+        );
+        selectorsInManagerFacet[3] = getSelector(
+            "setIncentiveToUAD(address)"
+        );
+        selectorsInManagerFacet[4] = getSelector(
+            "getExcessDollarsDistributor(address)"
+        );
+        selectorsInManagerFacet[5] = getSelector(
+            "initialize(address)"
+        );
+
+        bytes4[] memory fromGenSelectors = selectorsInManagerFacet;
+
+        // array of functions to add
+        FacetCut[] memory facetCut = new FacetCut[](1);
+        facetCut[0] = FacetCut({
+            facetAddress: address(managerFacet),
+            action: FacetCutAction.Add,
+            functionSelectors: fromGenSelectors
+        });
+
+        // add functions to diamond
+        vm.startPrank(owner);
+        ICut.diamondCut(facetCut, address(0x0), "");
+		vm.stopPrank();
+
+        ManagerFacet(address(diamond)).initialize(owner);
+    }
+}
+
+abstract contract CacheBugSetup is DiamondSetup {
+
+    ManagerFacet managerFacet;
+
+    bytes4 ownerSel = hex'8da5cb5b';
+    bytes4[] selectors;
+
+    function setUp() public virtual override {
+        super.setUp();
+        managerFacet = new ManagerFacet(owner);
+
+        selectors.push(hex'19e3b533');
+        selectors.push(hex'0716c2ae');
+        selectors.push(hex'11046047');
+        selectors.push(hex'cf3bbe18');
+        selectors.push(hex'24c1d5a7');
+        selectors.push(hex'cbb835f6');
+        selectors.push(hex'cbb835f7');
+        selectors.push(hex'cbb835f8');
+        selectors.push(hex'cbb835f9');
+        selectors.push(hex'cbb835fa');
+        selectors.push(hex'cbb835fb');
```

Comment:
I am curious how you generated hex values for selectors. It doesn't make sense to use hex values directly. 
you can update it as I mentioned above.

Reply:
![image](https://user-images.githubusercontent.com/4975670/201725255-c0a492ff-be7f-4161-9aa4-ba856032d025.png)

Reply:
> I am curious how you generated hex values for selectors. It doesn't make sense to use hex values directly. you can update it as I mentioned above.

I have added PrintSelectors.ts for checking the function selectors in any Facet
contracts/scripts/deploy/diamond/PrintSelectors.ts